engine/resource/src/providers/provider_archive.cpp

```diff
@@ -157,7 +157,7 @@ namespace dmResourceProviderArchive
 
     static dmResourceProvider::Result Unmount(dmResourceProvider::HArchiveInternal archive)
     {
-        delete (GameArchiveFile*)archive;
```

Comment:
We have to `UnmountArchiveInternal` otherwise calling of `mmap` without `munmap` maybe a reason of failure after one or a few reboots:
```
18:29:52.870  E  mmap(1581056,566935250) failed: Out of memory
18:29:52.870  W  create map from entry failed
18:29:52.870  E  ERROR:RESOURCE: Error when mapping data file 'game.arcd', result = -3
18:29:52.870  E  ERROR:RESOURCE: Failed to mount archive from 'game.arci' and 'game.arcd': IO_ERROR
18:29:52.870  E  ERROR:RESOURCE: Failed to mount base archive: -1000 for mount archive://game.dmanifest
```